tests/31sync/03joined.pl

```diff
@@ -34,6 +34,21 @@
          my $room = $body->{rooms}{joined}{$room_id};
          (!defined $room) or die "Unchanged rooms shouldn't be in the sync response";
 
+         # do another sync, with full_state
```

Comment:
Is this test intended to check interaction with the previous sync above? If not, then perhaps it would be best in its own test, for independence.